### PR TITLE
feat(surrealdb): use operator certification in agent readiness

### LINKED ISSUE #2801: [PHASE-2][SURREALDB][SLICE-5] Operator certification usage
## Mission
Operationalize operator certification usage for Phase 2 (extends #2776 closeout into repeatable adoption).

## Parent / lineage
- Epic: #2778
- Grandparent: #1976 (remains OPEN)
- Prerequisite evidence: #2776 CLOSED

## Scope
- Wire certification run into Phase-2 operator workflows (doctor/certify paths)
- PASS/WARN/FAIL matrix usage documented for adoption decisions
- No required CI dependency on live SurrealDB

## Not authorized
- LR/Live-Go, Echtgeld-Go
- productive SurrealDB writes; no default `PERSIST_ALLOWED=True` / `MUTATION_ALLOWED=True`
- trading state, secrets, orders, fills, positions, risk-state in SurrealDB
- raw SurrealQL exposed to agents
- automatic issue creation or code changes from Context Brain findings

## Validation
- Manual certification runbook path verified on Windows operator surface
- Redacted proof artifacts; failure blocks adoption claims but not unrelated PRs

## Activation note
Created under Jannek GO for #2778 planning phase (2026-06-02). Implementation requires scoped PR(s) with evidence.

### PR DESCRIPTION
**Pull request by jannekbuengener**
## Summary
- Integrate optional `operator_certification` / `context_certification` bundle field into `evaluate_agent_os_readiness_v1` (in-memory, no DB/network).
- Map context-certify proof-pack semantics to readiness findings: PASS/WARN/SKIPPED → weak or none; FAIL/BLOCKED → blocking adoption claims.
- Missing certification is reported in `missing_inputs` only (does not block unrelated PR readiness).
- Document operator flow and adoption matrix in SurrealDB MCP runbook + Wave-20 readiness runbook.

Closes #2801
Refs #2778
Refs #1976
Refs #2776

## Brain Evidence
brain_source: repo-only
brain_status: not-used
tools_or_queries:
  - pytest tests/unit/surrealdb/test_agent_os_readiness.py
  - pytest tests/unit/tools/mcp/test_agent_os_readiness_tools.py
  - pytest tests/unit/surrealdb/test_context_certify.py
  - make context-certify
records_or_results:
  - 77 + 7 unit tests passed locally
  - make context-certify: final_verdict=certified, tool_count=27, LR NO-GO
repo_crosscheck:
  - tools/surrealdb/agent_os_readiness.py (_evaluate_operator_certification)
  - tools/surrealdb/context_certify.py (CertifyReport shape)
impact_on_plan:
  - Evaluator-only integration; MCP adapter unchanged except guardrails count (6)
limitations:
  - No surrealdb-local MCP query; bundle must be supplied by operator/agent

## Delivered
- `_evaluate_operator_certification` in agent_os_readiness.py
- 9 new unit tests + MCP passthrough test
- Adoption matrix docs (#2801)

## Validation
- `git diff --check`
- `pytest -q tests/unit/surrealdb/test_agent_os_readiness.py tests/unit/tools/mcp/test_agent_os_readiness_tools.py` (77 passed)
- `pytest -q tests/unit/surrealdb/test_context_certify.py` (7 passed)
- `make context-certify` → certified, exit 0

## Test plan
- [x] Unit tests for certified/fail/blocked/skipped/missing/invalid certification
- [x] MCP handler passes certification through bundle
- [ ] CI required checks green

## Non-goals
- No CI gate on live SurrealDB
- No MCP/registry/permission_guard changes
- No Phase-2 slices #2798–#2804

## Safety Boundaries
- LR remains NO-GO; certification is adoption gate, not LR-Go
- PERSIST_ALLOWED=False, MUTATION_ALLOWED=False
- No productive DB writes, no runtime/BLUE/RED changes

## Restunsicherheiten
- Skipped checks always produce weak findings when present (by design for adoption documentation)

Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Medium Risk**
> Changes how readiness levels and adoption-blocking findings are derived from bundle input; behavior is read-only and well-tested but affects operator adoption signals.
> 
> **Overview**
> Wires optional **`operator_certification`** / **`context_certification`** (subset of `make context-certify` proof) into **`evaluate_agent_os_readiness_v1`**, still in-memory with no DB or network.
> 
> **Certification semantics:** FAIL/BLOCKED verdicts, blocked checks, or blocking gate failures → **blocking** findings (adoption claims blocked). WARN-style cases (non-blocking gate fails, skipped checks, invalid/incomplete proof) → **weak** findings plus extra **`required_validation`**. Absent certification is only **`missing_inputs`** and does **not** downgrade readiness from missing core bundle fields alone.
> 
> Adds a **sixth guardrail** (certification is an adoption gate, not LR-Go), documents the operator flow and adoption matrix in the SurrealDB MCP and Wave-20 runbooks, and extends unit/MCP tests for pass-through and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 69d68805edae21b40d9766a41e096fc55252c586. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by jannekbuengener** [comment]
Addressed P2 review in 69d68805:

1. **Require proof before pass** — incomplete check no longer treats \doption_status\ as substitute for CertifyReport proof; \doption_status=pass\ without \inal_verdict\ now yields weak + required_validation.
2. **WARN validation steps** — non-blocking gate failures and \doption_status=warn\ now append operator checklist entries to \
equired_validation\.

Tests: \	est_operator_certification_adoption_pass_without_proof_not_strong\, extended warn gate assertion.

File: tools/surrealdb/agent_os_readiness.py [hunk 1]
```diff
@@ -406,6 +413,154 @@ def _evaluate_architect_signals(
...
+    for gate in gate_matrix:
+        if not isinstance(gate, Mapping):
+            continue
+        status = _as_str(gate.get("status", "")).lower()
+        check_id = _as_str(gate.get("check_id", "?"))
+        detail = _as_str(gate.get("detail", ""))
+        if status == "fail" and gate.get("blocking") is not True:
+            weak.append(
+                f"operator certification gate warn (non-blocking fail): {check_id}"
+                + (f" — {detail}" if detail else "")
```
[the remark this answers is not in the document]
Comment (jannekbuengener) [reply tools/surrealdb/agent_os_readiness.py:526]: Fixed in 69d68805: non-blocking gate failures now append required_validation. adoption_status=warn gets the same. test_operator_certification_warn_non_blocking_gate asserts this.

File: tools/surrealdb/agent_os_readiness.py [hunk 2]
```diff
@@ -394,18 +405,153 @@ def _evaluate_architect_signals(
...
+    if adoption_status == "skipped":
+        weak.append("operator certification adoption_status=skipped")
+        if not extra_validation:
+            extra_validation.append("Document skip reasons before adoption claims.")
+
+    if not final_verdict and not adoption_status and not gate_matrix and not weak:
+        weak.append("operator certification incomplete: missing final_verdict")
+        extra_validation.append(
+            "Include final_verdict from make context-certify output."
+        )
```
[the remark this answers is not in the document]
Comment (jannekbuengener) [reply tools/surrealdb/agent_os_readiness.py]: Fixed in 69d68805: incomplete check no longer treats adoption_status as CertifyReport proof. adoption_status=pass without final_verdict yields weak + required_validation. test_operator_certification_adoption_pass_without_proof_not_strong.
